pkg/util/slices/slices.go

```diff
@@ -82,3 +82,14 @@ func CmpNoOrder[E comparable, S ~[]E](a, b S) bool {
 	}
 	return true
 }
+
+// Pick creates a new slice containing only the elements for which keep return true.
+func Pick[E any, S ~[]E](s S, keep func(*E) bool) S {
```

Comment:
Maybe we should rename this function to `Filter` because it's more common for this kind of operation.

From the https://github.com/kubernetes/utils/blob/e7106e64919e62d73c0926352134656a22dc2e38/strings/slices/slices.go#L39C1-L52C1

Reply:
In my opinion filter sold have an effect on the input, it's not the case here.